packages/idealo.design/scripts/motifuiImporter.js

```diff
@@ -29,62 +21,134 @@ async function readDirectory(directory) {
+                const endIndex = contentOfStoryFile.indexOf(', module')
+                component.screenshotFolderName = contentOfStoryFile.substring(endIndex - 1, startIndex + 11)
+                delete component.pathToStoryFile
+            }
         })
-    })
-
-    req.on('error', error => {
-        console.error(error)
-    })
-
-    req.write(JSON.stringify(data))
-    req.end()
+    }
+    return components
 }
 
-async function handleImportProcess(source, destination){
-    await fse.remove(destination)
-        .then(()=> console.log('delete folder successfully'))
-        .catch(err => console.log(err))
-    await fse.copy(source, destination)
-        .then(()=>console.log('copy folder successfully'))
-        .catch(err => console.log(err))
-    readDirectory(destination)
-        .then((result)=>readPackageJsons(result, destination)).then((result)=>sendDataToHttpRequest(result))
+async function storePathToScreenshots(components){
+    for(const component of components){
+        await fs.promises.readdir(localScreenshots + '/' + component.screenshotFolderName).then((screenshots) => {
+            component.screenshots = screenshots
+        })
+    }
+    return components
 }
 
-if(dangerousUpdateModeArgument){
-    handleImportProcess(pathToMotifUiRepo, motifUiFolder).then(()=>console.log('process finished'))
-}else{
-    console.error('something went wrong')
-}
+async function createFormDataForComponent(components) {
+    for (const component of components) {
+
+        await fs.mkdirSync(localPathToMotifUIScreenshots+'/'+component.screenshotFolderName, (err) => {
+            if (err) {
+                throw err;
+            }
+        });
+        const componentFormData = new FormData();
 
-module.exports.modeArgument = dangerousUpdateModeArgument
+        componentFormData.append('name', component.name);
+        for (const keyword of component.keywords) {
+            componentFormData.append('keywords', keyword);
+        }
 
+        componentFormData.append('screenshotFolderName', component.screenshotFolderName);
+        componentFormData.append('readme', component.readme);
 
+        for (const screenshot of component.screenshots) {
+            const screenshotBuffer = await fs.readFileSync(localScreenshots + '/' + component.screenshotFolderName + '/' + screenshot);
+            const screenshotName= screenshot.replace(/ /g, "_");
+            componentFormData.append('screenshots', screenshotBuffer, screenshotName);
+        }
+        await sendDataToHttpRequest(componentFormData)
+    }
+}
 
+async function sendDataToHttpRequest(componentFormData) {
+    return await axios.put('http://localhost:8080/api/components/update', componentFormData, {
+        headers: componentFormData.getHeaders()
+    });
+}
 
+async function handleImportProcess() {
+    await fse.remove(motifUiFolder)
+        .then(() => console.log('delete motifUI folder successfully'))
+        .catch(err => console.log(err))
```

Comment:
this must not be in the importer! You are changing files on another app (the server) that this app (the importer) does not have access to. The deletion of old screenshots on the server has to be performed by the server, not the importer.

Think of it this way: You, the server, order a new water filter for your coffee machine at Amazon. Amazon delivers it via a courier (the importer). The importer goes to your house with the payload (hitting the /components/update endpoint) and hands you the new water filter.

Then you (the server) will take the new filter, go to the coffee machine, throw away the old filter and put in the new one.

But what you are doing here in this code is allow the courier to enter your home and put their hand into your coffee machine to do god knows what :D please don't let them XD